### PR TITLE
Use opmath_type for CUDA logcumsumexp

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #83425

This improves precision by reducing the number of narrowing
conversions, as well as reducing compile times from 2m 30s to 1m 25s
on my machine.

### CODE REVIEW

File: aten/src/ATen/native/cuda/LogcumsumexpKernel.cu [hunk 1]
```diff
@@ -18,10 +18,11 @@ void launch_logcumsumexp_cuda_kernel(const TensorBase& result, const TensorBase&
       [&]() {
         using accscalar_t = acc_type<scalar_t, true>;
```
Comment (ngimel) [inline aten/src/ATen/native/cuda/LogcumsumexpKernel.cu:19]: better to use opmath_t here

**Review by ngimel — APPROVED** [review]

**Comment by huydhn** [comment]
@ngimel I'm trying to revert this PR because this seems to break rocm tests on trunk

```
======================================================================
FAIL [0.060s]: test_logcumsumexp_cuda (__main__.TestTorchDeviceTypeCUDA)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/conda/lib/python3.7/site-packages/torch/testing/_internal/common_utils.py", line 1921, in wrapper
    method(*args, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/torch/testing/_internal/common_utils.py", line 1921, in wrapper
    method(*args, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/torch/testing/_internal/common_device_type.py", line 377, in instantiated_test
    result = test(self, **param_kwargs)
  File "/opt/conda/lib/python3.7/site-packages/torch/testing/_internal/common_utils.py", line 988, in wrapper
    fn(*args, **kwargs)
  File "test_torch.py", line 2486, in test_logcumsumexp
    self.assertEqual(expected, actual)
  File "/opt/conda/lib/python3.7/site-packages/torch/testing/_internal/common_utils.py", line 2418, in assertEqual
    msg=(lambda generated_msg: f"{generated_msg} : {msg}") if isinstance(msg, str) and self.longMessage else msg,
  File "/opt/conda/lib/python3.7/site-packages/torch/testing/_comparison.py", line 1093, in assert_equal
    raise error_metas[0].to_error(msg)
AssertionError: Tensor-likes are not close!

Mismatched elements: 2 / 9 (22.2%)
Greatest absolute difference: nan at index (7,) (up to 1e-05 allowed)
Greatest relative difference: nan at index (7,) (up to 1.3e-06 allowed)
```

**Pull request closed without merging**